config/jobs/kubernetes-incubator/service-catalog/service-catalog-presubmits.yaml

```diff
@@ -18,3 +18,28 @@ presubmits:
         env:
         - name: NO_DOCKER
           value: "1"
+  - name: pull-service-catalog-xbuild
+    agent: kubernetes
+    context: pull-service-catalog-xbuild
+    always_run: true
+    skip_report: false
+    rerun_command: "/test pull-service-catalog-xbuild"
+    trigger: "(?m)^/test (all|pull-service-catalog-xbuild)\\s*"
+    spec:
+      containers:
+      - args:
+        - "--repo=sigs.k8s.io/$(REPO_NAME)=$(PULL_REFS)"
+        - "--root=/go/src/github.com/kubernetes-incubator/$(REPO_NAME)"
+        - "--timeout=90"
+        - "--scenario=execute"
+        - "--"
+        - "make"
+        - "images-all"
+        - "svcat-all"
+      env:
+      - name: DOCKER_IN_DOCKER_ENABLED
```

Comment:
I'm going to turn this into a preset soon. Initially we didn't because we wanted to discourage depending on it (I only initially built the support to help us get the rest of our existing jobs onto Prow), but so far it works just fine.

This must be paired with an emptyDir volume for the the graph storage (which is remapped `/var/lib/docker` for :reasons:, we should probably unset the option that changes that). Unless things have improved this is necessary to avoid issues with overlays on overlays. Existing docker-in-docker jobs should have this.

Reply:
What is a preset?

I was basing off of, which has preset-service-account
https://github.com/kubernetes/test-infra/blob/68c68683a066e7b9986113c73a6dcb0f48775ef9/config/jobs/kubernetes-sigs/cluster-api-provider-openstack/cluster-api-provider-openstack-presubmits.yaml

It does not have a emptydir, so maybe it's a bad example, oops.

Reply:
searched for `emptyDir` and found helm/charts repo
https://github.com/kubernetes/test-infra/blob/1c40d5125cd571010e6949fe0b41065f5973542c/config/jobs/helm/charts/charts.yaml

Reply:
preset is something we are mimicing https://kubernetes.io/docs/concepts/workloads/pods/podpreset/